File: internal/bindings/column.go
package bindings

import (
	"gomyadm/internal/models"
	"gomyadm/internal/services"
)

type ColumnBinding struct {
	service *services.ColumnService
}

func NewColumn(s *services.ColumnService) *ColumnBinding {
	return &ColumnBinding{service: s}
}

func (a *ColumnBinding) GetAll(table string) (*models.Table, error) {
	return a.service.GetAll(table)
}

func (a *ColumnBinding) Create(column models.ColumnRequest) error {
	return a.service.Create(column)
}

func (a *ColumnBinding) Update(column models.ColumnRequest) error {
	return a.service.Update(column)
}

func (a *ColumnBinding) Delete(table, column string) error {
	return a.service.Delete(table, column)
}

File: frontend/src/repositories/wails/column.ts
import { GetAll , Create, Update, Delete } from '../../../wailsjs/go/bindings/ColumnBinding'
import type { Result } from "..";
import type { ColumnRepository } from '../column';

export class WailsColumnRepository implements ColumnRepository {
  async getAll(table: string): Promise<Result<any>> {
    try {
      return {
        ok: true, data: await GetAll(table)
      }
    } catch (error) {
      return {
        ok: false, error: error instanceof Error ? error.message : "Erro ao conectar"
      }
    }
  }

  async create(column: any): Promise<Result<void>> {
    try {
      await Create(column)
      return {
        ok: true, data: undefined
      }
    } catch (error) {
      return {
        ok: false, error: error instanceof Error ? error.message : "Erro ao conectar"
      }
    }
  }

  async update(column: any): Promise<Result<void>> {
    try {
      await Update(column)
      return {
        ok: true, data: undefined
      }
    } catch (error) {
      return {
        ok: false, error: error instanceof Error ? error.message : "Erro ao conectar"
      }
    }
  }

  async delete(table: string, column: string): Promise<Result<void>> {
    try {
      await Delete(table, column)
      return {
        ok: true, data: undefined
      }
    } catch (error) {
      return {
        ok: false, error: error instanceof Error ? error.message : "Erro ao conectar"
      }
    }
  }
}